### PR TITLE
Allow specifying whether `targetValue` should be recursed

### PR DESCRIPTION
**Pull request by Richienb**
Based on: #25 

Fixes: #23, Fixes: #25

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -32,12 +32,18 @@ Source object to copy properties from.
 
 #### mapper
 
-Type: `Function`
+Type: `(sourceKey, sourceValue, source) => [targetKey, targetValue, mapperOptions?]`
```
Comment (Richienb) [inline readme.md:35]: TIL, a question mark after an item in a tuple will make the item optional. https://www.typescriptlang.org/docs/handbook/2/objects.html#tuple-types

If this change requires a newer version of TypeScript, we may as well use [named tuple elements](https://github.com/Microsoft/TypeScript/issues/28259#issuecomment-452912594).

Comment (sindresorhus) [reply readme.md:35]: > If this change requires a newer version of TypeScript, we may as well use named tuple elements.

👍

**Comment by sindresorhus** [comment]
I'm a little bit conflicted about whether it makes sense to add both this and https://github.com/sindresorhus/map-obj/pull/24. Any thoughts on that?

**Comment by Richienb** [comment]
This PR should make #24 mostly moot.

**Comment by Richienb** [comment]
Because of the new TypeScript feature, I needed to update `tsd` for it to work which added a new failing check unrelated to this PR:

```sh
  index.test-d.ts:33:0
  ✖  33:0  Parameter type { [key: string]: unknown; } is declared too wide for argument type { bar: "baz"; } & { [key: string]: unknown; }.  
```

**Pull request merged**